qos/evm/check.go

```diff
@@ -6,36 +6,39 @@ import (
 	"github.com/buildwithgrove/path/qos/jsonrpc"
 )
 
+// endpointStore provides the endpoint check generator required by
+// the gateway package to augment endpoints' quality data,
+// using synthetic service requests.
+var _ gateway.QoSEndpointCheckGenerator = &endpointStore{}
```

Comment:
Not a blocker: please add a TODO that endpointStore should be under serviceState (and not the other way round).

Reply:
Done.

Reply:
> Not a blocker: please add a TODO that endpointStore should be under serviceState (and not the other way round).

For clarity in implementing this change: where would the func `func (es *endpointStore) filterValidEndpoints(availableEndpoints []protocol.EndpointAddr) ([]protocol.EndpointAddr, error)` live?